win32ss/gdi/gdi32/objects/gdiobj.c

```diff
@@ -265,10 +265,11 @@ GetObjectW(
                 (dwType == GDI_OBJECT_TYPE_BRUSH) ||
                 (dwType == GDI_OBJECT_TYPE_COLORSPACE) ||
                 ( (dwType == GDI_OBJECT_TYPE_EXTPEN) &&
-                    ( (cbSize >= sizeof(EXTLOGPEN)) || (cbSize == 0) ) ) ||
+                    ( (cbSize >= sizeof(EXTLOGPEN)))) ||
                 ( (dwType == GDI_OBJECT_TYPE_BITMAP) && (cbSize >= sizeof(BITMAP)) ))
             {
-                SetLastError(ERROR_NOACCESS);
+                if (cbSize)
```

Comment:
So what's the error set when cbSize == 0? none (stays as the ambient one)?

Reply:
This is the result observed in Win2003Srv, but I suspect it should be ERROR_INVALID_PARAMETER, as the function returns 0 indicating an error condition.
Win11 has a different result (with the original code).

![CapturaWin11GetObject](https://github.com/reactos/reactos/assets/49964946/26c0e7fa-1a10-4040-a263-1482da8e766e)

Reply:
Yes, this is the behavior of Win2003 and WinXP. Only Win11 has a different behavior, as in the present case, with a válid HBRUSH hand
![CapturaWin11GetObject](https://github.com/reactos/reactos/assets/49964946/dbe7548b-53ed-42f8-80ca-4f57ad135297)
ler, szSize = 0 and a válid value in the buffer address the function returns the sizeof(LOGBRUSH) == 0x0c.